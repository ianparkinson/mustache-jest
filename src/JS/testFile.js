const template = require('./testTemplate.html').default;

/**
 *
 * @param {HTMLDivElement} div
 * @param {object} data
 */
module.exports = function renderToDiv(div, data) {
  const htmlCompiled = template(data);

  div.innerHTML = htmlCompiled;
};
