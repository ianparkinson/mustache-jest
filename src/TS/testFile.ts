import template from './testTemplate.html';

export function renderToDiv(div: HTMLDivElement, data: object) {
  const htmlCompiled = template(data);

  div.innerHTML = htmlCompiled;
}
