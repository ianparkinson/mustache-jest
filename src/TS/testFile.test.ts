import { renderToDiv } from './testFile';

describe('renderToDiv', () => {
  let divElement: HTMLDivElement;

  beforeEach(() => {
    divElement = document.createElement('div');
    renderToDiv(divElement, { name: 'josie', shouldShow: true });
  });

  it('contains the content from the compiled template', () => {
    expect(divElement.innerHTML).toMatchSnapshot();
  })
});
