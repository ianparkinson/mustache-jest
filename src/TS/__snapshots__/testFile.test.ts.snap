// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderToDiv contains the content from the compiled template 1`] = `
"<div>
  <p>Hello josie!</p>
  <div>Welcome!</div>
</div>
"
`;
